### PR TITLE
Image for truncate metrics and tag indices when large partition happens.

### PR DESCRIPTION
**Pull request by rubenvp8510**
Not sure if this is the best way to do it.

@jmartisk WDYT?

I did modifications[1] to openshift-ansible installer, this isfor install a kubernetes cronjob to be executed each week (configurable) 

1) https://github.com/rubenvp8510/openshift-ansible/pull/5

### CODE REVIEW

File: truncator-cronjob/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+FROM openshift/origin:latest
+
+# The image is maintained by the Hawkular Metrics team
+MAINTAINER Hawkular Metrics <hawkular-dev@lists.jboss.org>
+
+# Environment variables exposed.
+ENV PATH=$PATH:/usr/local/bin \
+    HAWKULAR_NAMESPACE="openshift-infra" \
+    CASSANDRA_CONTAINER_PREFIX="hawkular-cassandra" \
+    PARTITION_THRESHOLD=157286400
```
Comment (jmartisk) [inline truncator-cronjob/Dockerfile:29]: should this be configurable?

File: truncator-cronjob/truncate-idx.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -e
+
+# We will see if we required to truncate tables, if we found a large partition (larger than the threshold)
+# proceeed to truncate tables and recreate indices.
+
+# Get one cassandra pod so we can execute nodetool
+CASSANDRA_POD=$(oc get pods -n ${HAWKULAR_NAMESPACE} | grep hawkular-cassandra | head -n 1 |  awk '{print $1}')
```
Comment (jmartisk) [inline truncator-cronjob/truncate-idx.sh:8]: Perhaps, just to be sure, this should check that it picks a Cassandra pod that is in Running state so we don't grab one that is being started/shutting down at that moment. And fail with a reasonable message if there's no suitable pod.

File: truncator-cronjob/truncate-idx.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+
+# Get largest partition of metrics_idx and metrics_tags_idx
+MAX_PARTIITON_TAGS_IDX=$(oc -n ${HAWKULAR_NAMESPACE} exec ${CASSANDRA_POD} -- nodetool tablehistograms hawkular_metrics metrics_tags_idx | grep Max | awk '{print $5}')
```
Comment (jmartisk) [inline truncator-cronjob/truncate-idx.sh:12]: there's a typo in these variables (PARTIITON)

File: truncator-cronjob/truncate-idx.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+if ([ "$MAX_PARTIITON_TAGS_IDX" != "NaN" ] && [ $MAX_PARTIITON_TAGS_IDX -gt $PARTITION_THRESHOLD ]) || ([ "$MAX_PARTIITON_IDX" != "NaN" ] && [ $MAX_PARTIITON_IDX -gt $PARTITION_THRESHOLD ]); then
+  echo "Scalling down Heapster and Hawkular pods"
+  oc -n ${HAWKULAR_NAMESPACE} scale rc heapster --replicas=0
+  oc -n ${HAWKULAR_NAMESPACE} scale rc hawkular-metrics --replicas=0
+  echo "Cassandra pod is ${CASSANDRA_POD}. Truncating tables"
+  oc -n ${HAWKULAR_NAMESPACE} exec ${CASSANDRA_POD} -- cqlsh --ssl -e "truncate table hawkular_metrics.metrics_tags_idx"
+  oc -n ${HAWKULAR_NAMESPACE} exec ${CASSANDRA_POD} -- cqlsh --ssl -e "truncate table hawkular_metrics.metrics_idx"
+  echo "Scalling up Hawkular and Heapster pods"
+  oc -n ${HAWKULAR_NAMESPACE} scale rc hawkular-metrics --replicas=1
```
Comment (jmartisk) [inline truncator-cronjob/truncate-idx.sh:25]: Shouldn't we find out how many replicas there were before and return them to the same number?

**Comment by jmartisk** [comment]
Found a few potential issues, otherwise I think it's ok..

File: truncator-cronjob/truncate-idx.sh (hunk 3 above)
Comment (rubenvp8510) [reply truncator-cronjob/truncate-idx.sh:12]: Nice catch, thanks

File: truncator-cronjob/truncate-idx.sh (hunk 4 above)
Comment (rubenvp8510) [reply truncator-cronjob/truncate-idx.sh:25]: Yes!

File: truncator-cronjob/truncate-idx.sh (hunk 2 above)
Comment (rubenvp8510) [reply truncator-cronjob/truncate-idx.sh:8]: Yes, you're right! I'll add a check here.

File: truncator-cronjob/Dockerfile (hunk 1 above)
Comment (rubenvp8510) [reply truncator-cronjob/Dockerfile:29]: It is, that is only a default value, you can override this value.

Comment (jmartisk) [reply truncator-cronjob/Dockerfile:29]: well you override it by the `hawkular_truncation_cron.j2` template but there's also a hardcoded value [1] (and a different one than here, which is confusing), perhaps the template should refer to an ansible variable that you can set for the installation?

[1] https://github.com/rubenvp8510/openshift-ansible/pull/5#discussion-diff-276544518R34

**Pull request closed without merging**